packages/file-pipeline/src/helpers/agnostic-source/agnostic-source.test.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs'
+import {resolve} from 'path'
+import {agnosticSource} from '.'
+import {testStreamItems} from '../../test-utils'
+
+const cwd = resolve(__dirname, 'fixtures')
+
+function logItem(fileOrString: {path: string} | string) {
+  if (typeof fileOrString === 'string') {
+    return fileOrString
+  }
+  return fileOrString.path
+}
+
+describe('agnosticSource', () => {
+  afterEach(() => {
+    if (fs.existsSync(resolve(cwd, 'three'))) {
+      fs.unlinkSync(resolve(cwd, 'three'))
+    }
+  })
+
+  test('when watching = false', (done) => {
+    const expected = [resolve(cwd, 'one'), resolve(cwd, 'two')]
+    const {stream} = agnosticSource({ignore: [], include: ['**/*'], cwd, watch: false})
+    const log: any[] = []
+    stream.on('data', (data) => {
+      if (data === 'ready') {
+        stream.end()
+        expect(log).toEqual(expected)
+        return done()
+      }
+      log.push(data.path)
+    })
+  })
+
+  test.skip('when watching = true', async () => {
```

Comment:
Is there a specific reason this test is skipped?

Reply:
Nice catch!